### DISCUSSION TITLE
questions on annotation

### DISCUSSION THREAD

**Discussion by Lovkush-A** [alan-turing-institute/sktime#1520] [Q&A]
1. Is annotation (as currently defined in the base class) just a special case of forecasting with exogenous variables? I explain why I think this below:

From the base class of annotation:
* `Assumes "predict" data is temporal future of "fit"`
* `fitting              - fit(self, X, Y=None)`
* `annotating           - predict(self, X)`

I think this can be rephrased as: forecasting Y using exogenous variables X.

2. Are we missing the case where Y can be categorical? Currently we have labels "indicator" for boolean and "score" for float.

3. Are there plans to consider this alternative version of annotation:
`X` is panel data, so a *list* of dataframes, where each dataframe represents a time series. `Y` is a list of dataframes, where `Y[i]` is the annotation for `X[i]`.  Then in predict, `X_new`  is new panel of data, and we want to predict Y for each series in `X_new`

**Comment by fkiraly** [comment]
>I think this can be rephrased as: forecasting Y using exogenous variables X.

Except in general there is no "y" in `fit`?

>Are we missing the case where Y can be categorical? 

Categorical should also be allowed.

>Are there plans to consider this alternative version of annotation:

Yes, there is this old PR: https://github.com/alan-turing-institute/sktime/pull/1036

which Markus asked me to split off the stream annotator, and then went on ignoring.

**Reply by Lovkush-A** [reply]
> Except in general there is no "y" in `fit`?

Interesting!

4. Does that mean all concrete classes have to be able to do handle both Y is None and Y is not None? Or can concrete classes specify what kind of Y it can handle?

5. For the Y is not None case, looks like we have a reduction (to forecasting with exogenous variables) to implement!
